src/XMakeBuildEngine/Evaluation/ProjectRootElementCache.cs

```diff
@@ -194,7 +194,7 @@ internal ProjectRootElementCache(bool autoReloadFromDisk)
         /// <param name="preserveFormatting"><code>true</code> to the project was loaded with the formated preserved, otherwise <code>false</code>.</param>
         /// <returns>The ProjectRootElement instance if one exists.  Null otherwise.</returns>
         internal ProjectRootElement Get(string projectFile, OpenProjectRootElement openProjectRootElement, bool isExplicitlyLoaded,
-            bool preserveFormatting)
+            bool? preserveFormatting)
```

Comment:
If `preserveFormatting` is not specified then the creation of the project should default to `false`, as that is the previous behavior. However, your second idea sounds good as well. The goal of the changes to preserve formatting is to be able to load it once with the formatting defaulted *or* specified, but then any subsequent loads you have the option to say "don't care, give me what you have already loaded". Without the nullable bool behavior exposed, calls to `Open` and `TryOpen` cannot say they just want whatever is already loaded.